create/node_aws.go

```diff
@@ -152,11 +152,28 @@ func newAWSNode(selectedClusterManager, selectedCluster string, remoteBackend ba
 		cfg.AWSInstanceType = result
 	}
 
-	// Add new node to terraform config
-	err = state.Add(fmt.Sprintf(awsNodeKeyFormat, cfg.Hostname), &cfg)
+	// Get existing node names
+	nodes, err := state.Nodes(selectedCluster)
 	if err != nil {
 		return err
 	}
+	existingNames := []string{}
+	for nodeName := range nodes {
+		existingNames = append(existingNames, nodeName)
+	}
+
+	// Determine what the hostnames should be for the new node(s)
+	newHostnames := getNewHostnames(existingNames, cfg.Hostname, cfg.NodeCount)
+
+	// Add new node to terraform config with the new hostnames
+	for _, newHostname := range newHostnames {
+		cfgCopy := *(&cfg)
```

Comment:
Do we need this to create a copy of the struct?

I think `cfgCopy := cfg` would do the same. https://play.golang.org/p/XK_S0iQSnE

Reply:
Oh interesting. I didn't know go makes a copy when you assign structs like that. I'll fix it.

Reply:
Yup, the only time this doesn't work is if the struct contains reference types. (pointers, slice, etc)

If the struct contained reference types go would just copy the reference not the full value.